### ISSUE TITLE
wfreerdp.exe Segmentation fault connecting to Gnome remote desktop

### ISSUE THREAD

**Issue by jpmorrison** [FreeRDP/FreeRDP#7419]
wfreerdp connects to Gnome remote desktop, displays/updates the desktop, etc. 

crashes as soon as mouse enters the remote desktop.

mstsc works

```
wfreerdp.exe    /v:x.x.x.x
[16:30:21:199] [63180:0000b038] [INFO][com.freerdp.core] - freerdp_connect:freerdp_set_last_error_ex resetting error state
[16:30:21:199] [63180:0000b038] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[16:30:21:199] [63180:0000b038] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[16:30:21:199] [63180:0000b038] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[16:30:21:199] [63180:0000b038] [INFO][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
[16:30:21:226] [63180:0000b038] [INFO][com.freerdp.primitives] - primitives autodetect, using generic
[16:30:21:239] [63180:0000b038] [INFO][com.freerdp.core] - freerdp_tcp_is_hostname_resolvable:freerdp_set_last_error_ex resetting error state
[16:30:21:239] [63180:0000b038] [INFO][com.freerdp.core] - freerdp_tcp_default_connect:freerdp_set_last_error_ex resetting error state
[16:30:21:288] [63180:0000b038] [INFO][com.freerdp.crypto] - known_hosts=1
[16:30:27:059] [63180:0000b038] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[16:30:27:059] [63180:0000b038] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_BGRA32
[16:30:27:093] [63180:0000b038] [INFO][com.freerdp.channels.rdpsnd.client] - [static] Loaded fake backend for rdpsnd
[16:30:27:093] [63180:0000b038] [INFO][com.freerdp.channels.drdynvc.client] - Loading Dynamic Virtual Channel rdpgfx
[16:30:27:258] [63180:0000b038] [INFO][com.freerdp.client.windows] - Window is shown!
Segmentation fault


wfreerdp /buildconfig
This is FreeRDP version 3.0.0-dev (391f2f7c4)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H= HAVE_EXECINFO_H= HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_STRNDUP= HAVE_SYSLOG_H= HAVE_SYS_EVENTFD_H= HAVE_SYS_FILIO_H= HAVE_SYS_SELECT_H= HAVE_SYS_SOCKIO_H= HAVE_SYS_TIMERFD_H= HAVE_TM_GMTOFF= HAVE_UNISTD_H= WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_EVENTS=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=OFF WITH_FREERDP_DEPRECATED=OFF WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_ICU=OFF WITH_IPP=OFF WITH_JPEG=OFF WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_MEDIA_FOUNDATION=ON WITH_NATIVE_SSPI=ON WITH_OPENCL=OFF WITH_OPENH264=OFF WITH_OPENSSL=ON WITH_PROFILER=OFF WITH_PROGRESS_BAR=ON WITH_SAMPLE=OFF WITH_SERVER=OFF WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=OFF WITH_THIRD_PARTY=OFF WITH_VERBOSE_WINPR_ASSERT=ON WITH_WIN8=OFF WITH_WINDOWS_CERT_STORE=ON WITH_WINMM=ON WITH_WINPR_DEPRECATED=OFF WITH_WINPR_TOOLS=ON WITH_WIN_CONSOLE=OFF WITH_ZLIB=OFF
Build type:          Release
CFLAGS:              /DWIN32 /D_WINDOWS  /Gd /W3
Compiler:            MSVC, 18.0.40629.0
Target architecture: x64

```

**Comment by pnowack** [comment]
@akallabeth Very likely that wfreerdp accesses the `andMaskData` unconditionally (it should be NULL here).

**Comment by akallabeth** [comment]
@pnowack where should that happen? If `andMaskData == NULL` `lengthAndMask == 0` or do I miss something there?

**Comment by pnowack** [comment]
> If andMaskData == NULL lengthAndMask == 0 or do I miss something there?

Yes, with g-r-d `andMaskData == NULL` and `lengthAndMask == 0` are always true. I had to fix an issue in Remmina over a year ago, where Remmina went in a disconnect-reconnect loop, since it had a superfluous `NULL` check on the `andMaskData`.
Also OR mentions:

> crashes as soon as mouse enters the remote desktop.

In that case g-r-d sent the pointer data. g-r-ds pointer data is always 32bpp xorMaskData without andMaskData.

> where should that happen?

I did not check directly the code, but the assumption is based on these two indicators here.
